iis-wf/iis-wf-import/src/test/java/eu/dnetlib/iis/wf/importer/converter/ProjectToOrganizationConverterTest.java

```diff
@@ -0,0 +1,159 @@
+
+	private byte[] relationColumnFamilyBytes = "dummy".getBytes();
+
+	private Oaf oaf = null;
+
+	private Result result = mock(Result.class);
+
+	private ResultApprover defaultApprover = mock(ResultApprover.class);
+
+	@Before
+	public void setUp() {
+		when(defaultApprover.approveBeforeBuilding(any())).thenReturn(true);
+	}
+
+	// ------------------------ TESTS --------------------------
+
+	@Test(expected = NullPointerException.class)
+	public void constructor_relationColumnFamilyBytes_NULL() throws Exception {
+		new ProjectToOrganizationConverter(defaultApprover, null);
+	}
+
+	@Test(expected = NullPointerException.class)
+	public void constructor_resultApprover_NULL() throws Exception {
+		new ProjectToOrganizationConverter(null, relationColumnFamilyBytes);
+	}
+
+	@Test(expected = NullPointerException.class)
+	public void buildObject_hbaseResult_NULL() throws Exception {
+		// given
+		ProjectToOrganizationConverter converter = new ProjectToOrganizationConverter(defaultApprover,
+				relationColumnFamilyBytes);
+		// execute
+		converter.buildObject(null, oaf);
+	}
+
+	@Test
+	public void buildObject_disapproved_candidates() throws Exception {
+		// given
+		when(defaultApprover.approveBeforeBuilding(any())).thenReturn(false);
+		ProjectToOrganizationConverter converter = new ProjectToOrganizationConverter(defaultApprover,
+				relationColumnFamilyBytes);
+
+		String projectId = "someProjectId";
+		String organizationId = "someOrgId";
+
+		NavigableMap<byte[], byte[]> navigableMap = new TreeMap<byte[], byte[]>(
+				UnsignedBytes.lexicographicalComparator());
+		navigableMap.put(organizationId.getBytes(), createOafObject(projectId, organizationId).toByteArray());
+		when(result.getFamilyMap(relationColumnFamilyBytes)).thenReturn(navigableMap);
+
+		// execute
+		Collection<ProjectToOrganization> projOrgs = converter.buildObject(result, oaf);
+
+		// assert
+		assertNotNull(projOrgs);
+		assertTrue(projOrgs.isEmpty());
+	}
+
+	@Test
+	public void buildObject_no_candidates() throws Exception {
+		// given
+		ProjectToOrganizationConverter converter = new ProjectToOrganizationConverter(defaultApprover,
+				relationColumnFamilyBytes);
+
+		when(result.getFamilyMap(relationColumnFamilyBytes))
+				.thenReturn(new TreeMap<byte[], byte[]>(UnsignedBytes.lexicographicalComparator()));
+
+		// execute
+		Collection<ProjectToOrganization> projOrgs = converter.buildObject(result, oaf);
+
+		// assert
+		assertNotNull(projOrgs);
+		assertTrue(projOrgs.isEmpty());
+	}
+
+	@Test
+	public void buildObject() throws Exception {
+		// given
+		ProjectToOrganizationConverter converter = new ProjectToOrganizationConverter(defaultApprover,
+				relationColumnFamilyBytes);
+
+		String projectId = "someProjectId";
+		String organization1Id = "someOrg1Id";
+		String organization2Id = "someOrg2Id";
+
+		NavigableMap<byte[], byte[]> navigableMap = new TreeMap<byte[], byte[]>(
+				UnsignedBytes.lexicographicalComparator());
```

Comment:
Let's talk about it at our today's meeting. In my view breaking the lines in such a way makes it harder to read the code.